testing/src/main/java/io/stargate/it/http/ApiServiceParameters.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.it.http;
+
+import java.util.Collections;
+import java.util.Map;
+import org.immutables.value.Value;
+
+/**
+ * Defines parameters for each of the Service API instances requested by a test. Recommended usage:
+ * provide a base class for tests for each Service API that provides a method to customize these
+ * parameters as needed.
+ *
+ * @see ApiServiceSpec#parametersCustomizer()
+ * @see BaseRestApiTest#buildApiServiceParameters(Builder)
+ */
+@Value.Immutable(prehash = true)
+public interface ApiServiceParameters {
+
+  @Value.Default
+  default Map<String, String> systemProperties() {
+    return Collections.emptyMap();
+  }
+
+  @Value.Default
+  default String listenAddress() {
+    // Note: 127.0.1.N addresses are used by proxy protocol testing,
+    // 127.0.2.N addresses are used for Stargate coordinator nodes,
+    // so we allocate from the 127.0.3.X range here, to avoid conflicts with other
+    // services that may be listening on the common range of 127.0.0.Y addresses.
+    return "127.0.3.1";
+  }
+
+  @Value.Default
+  // Example: "REST API"
+  default String serviceName() {
+    return "Override me";
+  }
+
+  @Value.Default
+  // Example: 8088
+  default int servicePort() {
+    return 0;
+  }
+
+  @Value.Default
+  // Example: dw.server.connector.port
+  default String servicePortPropertyName() {
+    return "Override me";
```

Comment:
If this is truly required, could/should it be left with no implementation at all? (or maybe Immutables requires this?)

Reply:
I guess this was just refactored from the existing class with no changes.

Reply:
Good point. I found a better way to do this with `@Value.Parameter` which doesn't require a default value.